### ISSUE TITLE
Logging Not Registering In Flask 0.12

### ISSUE THREAD

**Issue by nnewman** [getsentry/raven-python#946]
I'm having some trouble getting logging to work with Raven. I noticed today that the handler gets set up properly if I run Flask 0.11.1, but not on 0.12. I made some example code below which is able to reproduce the issue. I've been running this on Python 3.5 with just Flask and Raven installed, Raven version 5.32.0.

### Symptoms

* Sentry events not reported for the below code, upon hitting the `/` route, using Flask 0.12, while events show for the same conditions using Flask 0.11.0

### Expected Outcome

* Sentry events should be reported across Flask versions

```python

import logging

from flask import Flask
from raven.contrib.flask import Sentry

sentry = Sentry()

def create_app():
    app = Flask(__name__)

    app.config['DEBUG'] = True

    sentry.init_app(
        app,
        dsn='https://<redacted>',
        logging=True,
        level=logging.WARNING
    )

    def internal_server_error(exception):
        app.logger.exception(exception)
        return 'Internal Server Error'

    app.errorhandler(Exception)(internal_server_error)

    return app


app = create_app()


@app.route('/')
def test_route():
    raise Exception('TEST')


if __name__ == '__main__':
    app.run()

```

**Comment by dcramer** [comment]
@mitsuhiko did you change something in Flask 0.12 around the logging bits?

**Comment by mitsuhiko** [comment]
This code above should never have worked. The correct way to register server error handler has always been `@app.errorhandler(500)` and that continues to work. It also always logs.

If you catch down an `Exception` it means the error is "handled" and the default error logging is disabled. I suppose this is why an attempt is made here to log with `app.logger.exception`. However since the exception is an object and not a string i suppose something might break.

I will have a look at this but the correct way is to use `app.errorhandler(500)` and from a quick look this works.

**Comment by nnewman** [comment]
It appears swapping out to using `app.errorhandler(500)` does in fact work as expected with Flask 0.12. In Flask 0.11.1, the way I posted above was totally functional with Sentry's logging module, hence the surprise when it stopped working under 0.12. Nevertheless, closing as this is resolved.

Thanks!

**Comment by nnewman** [comment]
@mitsuhiko - apparently I was too quick to close this. You are definitely correct regarding the error handler, however the hook into the Flask application logger still appears to be broken. I have updated my sample application and pasted below. Once again, on Flask v0.11.1, upon hitting `/log` I get an error in Sentry, however when I updated Flask to v0.12, I no longer saw an error appear in Sentry upon hitting that route.

```python
import logging

from flask import Flask, jsonify
from raven.contrib.flask import Sentry

sentry = Sentry()

def create_app():
    app = Flask(__name__)

    app.config['DEBUG'] = True

    sentry.init_app(
        app,
        dsn='https://<redacted>',
        logging=True,
        level=logging.DEBUG
    )

    @app.errorhandler(500)
    def internal_server_error(exception):
        return 'Internal Server Error'

    return app


app = create_app()


@app.route('/log')
def test_log():
    app.logger.error('TEST ERROR')
    return jsonify({'message': 'ok'})


@app.route('/')
def test_route():
    raise Exception('TEST')


if __name__ == '__main__':
    app.run()
```

**Comment by mitsuhiko** [comment]
Is there a specific reason you are running this in debug mode?

**Comment by nnewman** [comment]
Nope, just my default for development. Turning off `DEBUG` and restarting the app does not alter the behavior.

**Comment by mitsuhiko** [comment]
I wonder if we currently require blinker to be installed. I cannot reproduce your behavior here. I will try a bit later with latest release versions though.

**Comment by nnewman** [comment]
Interesting that you cannot reproduce. In case it helps, I'm using a bare virtualenv, with Python 3.5 on OSX 10.12.2, and have only explicitly installed Flask, and raven[flask] v5.32.0 in the virtualenv.

Below is a readout of all the packages/dependencies in the virtualenv:
```
$ pip3 freeze
blinker==1.4
click==6.7
contextlib2==0.5.4
Flask==0.12
itsdangerous==0.24
Jinja2==2.9.5
MarkupSafe==0.23
raven==5.32.0
Werkzeug==0.11.15
wheel==0.24.0
```

**Comment by jm66** [comment]
Hey, I'm experiencing the same issue but using python 2.7.13:

```
appdirs==1.4.0
blinker==1.4
click==6.7
contextlib2==0.5.4
Flask==0.12
itsdangerous==0.24
Jinja2==2.9.5
MarkupSafe==0.23
packaging==16.8
pyparsing==2.1.10
raven==5.32.0
six==1.10.0
Werkzeug==0.11.15
```

**Comment by jm66** [comment]
Tried setting `app.logger.propagate = True` and it worked as expected. Got it from https://github.com/pallets/flask/commit/e00e2c22aadf9e5b9dc353c31f1b70dad488b47a

**Comment by jdmwood** [comment]
I had the same issue in Raven 6.0.0 and Flask 0.12 (logging to app.logger doesn't propagate).

@jm66 's fix worked for me, however.
